# Sequenza Automation

# Manualë¡œ ì‘ì—… ìˆ˜í–‰í•˜ê¸°

```bash
conda activate sequenza - sequenza ì‹¤í–‰ì„ ìœ„í•œ env activation
sequenza-utils gc_wiggle -w 50 --fasta human_g1k_v37.fasta -o human_g1k_v37.wig.gz (/data/SJ/projects/flagship_figure/gc_wiggle/human_g1k_v37.wig.gz)
sequenza-utils bam2seqz -n normal.bam -t tumor.bam --fasta human_g1k_v37.fasta -gc human_g1k_v37.wig.gz -o out.seqz.gz
sequenza-utils seqz_binning --seqz out.seqz.gz -w 50 -o out small.seqz.gz
Rscript /data/SJ/projects/flagship_figure/ext_11a/scripts/run_sequenza.R -i small.seqz.gz -o output_dir -n sample_name
```

# Automation Pipeline

## ê°œë°œ ëª©ì 

- Sequenza í™˜ê²½ìœ¼ë¡œ ì¸í•œ ë¬¸ì œ
    - í•´ë‹¹ Toolì˜ ê²½ìš°, Rê³¼ Pythonì„ ëª¨ë‘ ì‚¬ìš©í•˜ë„ë¡ ì„¤ê³„ë˜ì–´ ìˆëŠ”ë°, ì´ ë•Œ ë‘ Toolì˜ ê°œë³„ì ì¸ í™˜ê²½ê³¼ ë”ë¶ˆì–´ Rê³¼ Python ê°„ì˜ ë²„ì „ì´ ë§ì•„ì•¼ í•¨ë„ ê³ ë ¤ë˜ì–´ì•¼ í•¨.
- ë§ì€ ìƒ˜í”Œì— ëŒ€í•´ ë¶„ì„ì„ ìˆ˜í–‰í•˜ê³ ì í•¨.
    - Clone ìƒ˜í”Œì— ëŒ€í•œ ë¶„ì„ì„ í•˜ê³ ì í–ˆì„ ë•Œ, ê°œìˆ˜ê°€ ë§ë‹¤ ë³´ë‹ˆ ì´ë¥¼ ìœ„í•œ ìë™í™”ê°€ í•„ìš”í•˜ë‹¤ê³  ìƒê°í•˜ì˜€ìŒ.
- Flagship Figure ì‘ì„± ì‹œ, Sequenzaì— ëŒ€í•œ ë¶„ì„ ê²°ê³¼ íŒŒì¼ì´ í•„ìš”.

## ê°œë°œ ê³¼ì •

- Snakemakeë¥¼ í†µí•œ ìë™í™”
- Dockerhubì—ì„œ ë°°í¬ ì¤‘ì¸ ê³µì‹ Imageë¥¼ í†µí•˜ì—¬ í™˜ê²½ ì˜ì¡´ì„± / í˜¸í™˜ ë¬¸ì œë¥¼ ìµœì†Œí™”
    
    ![image.png](image.png)
    
- í•„ìš” í™˜ê²½ì— ëŒ€í•œ Containerë¡œ ë™ì‘í•˜ë„ë¡ í•˜ì—¬ Cloud í™˜ê²½ì— ì˜¬ë ¤ë„ ë°œìƒí•  ìˆ˜ ìˆëŠ” ë¬¸ì œë¥¼ ìµœì†Œí™” í•˜ì˜€ìŒ.

# íŒŒì¼ ê²½ë¡œ
# ğŸ“‚ Pipeline Path
**`/data/SJ/pipeline/sequenza_pipeline`**  
ğŸ‘‰ ê³§ GitHub ë°±ì—… í›„ ì—°êµ¬ì‹¤ ë©”ì¸ ë””ë ‰í† ë¦¬ë¡œ ì´ê´€ ì˜ˆì •

```
sequenza_pipeline/
â”œâ”€â”€ Snakefile
â”œâ”€â”€ bin/
â”‚   â”œâ”€â”€ benchmarks_summary.py   # ì‹¤í–‰ ì‹œê°„, CPU ì‚¬ìš©ëŸ‰ ê³„ì‚° ìŠ¤í¬ë¦½íŠ¸
â”‚   â”œâ”€â”€ config.py               # Pipeline ì‹¤í–‰ ê´€ë ¨ Config ë¡œë“œ ìŠ¤í¬ë¦½íŠ¸
â”‚   â””â”€â”€ run_sequenza.R          # Sequenza ì‹¤í–‰ìš© R ìŠ¤í¬ë¦½íŠ¸
â”œâ”€â”€ envs/
â”‚   â””â”€â”€ cluster.json            # cluster í™˜ê²½ êµ¬ì„± (í˜„ì¬ëŠ” ë¯¸ì‚¬ìš©)
â”œâ”€â”€ modules/
â”‚   â””â”€â”€ sequenza.snakefile      # Sequenza Rule ì •ì˜
â””â”€â”€ refs/
    â””â”€â”€ ref.yaml                # Reference Genome ê²½ë¡œ (í˜„ì¬ GRCh37, ì¶”í›„ GRCh38 ì¶”ê°€ ì˜ˆì •)
```

---

# ğŸ“‚ Analysis Path
**`/data/SJ/projects/flagship_figure/analysis_pipeline`**  
ğŸ‘‰ ê³§ GitHub ë°±ì—… í›„ ì—°êµ¬ì‹¤ ë©”ì¸ ë””ë ‰í† ë¦¬ë¡œ ì´ê´€ ì˜ˆì •

```
analysis_pipeline/
â”œâ”€â”€ analysis/
â”‚   â””â”€â”€ 10_post_rt_leg_fb_3-2/           # ì˜ˆì‹œ ìƒ˜í”Œëª…
â”‚       â”œâ”€â”€ output/                      # ìµœì¢… ê²°ê³¼ë¬¼
â”‚       â”‚   â”œâ”€â”€ *_CN_bars.pdf
â”‚       â”‚   â”œâ”€â”€ *_CP_contours.pdf
â”‚       â”‚   â”œâ”€â”€ *_alternative_fit.pdf
â”‚       â”‚   â”œâ”€â”€ *_alternative_solutions.txt
â”‚       â”‚   â”œâ”€â”€ *_chromosome_depths.pdf
â”‚       â”‚   â”œâ”€â”€ *_chromosome_view.pdf
â”‚       â”‚   â”œâ”€â”€ *_confints_CP.txt
â”‚       â”‚   â”œâ”€â”€ *_gc_plots.pdf
â”‚       â”‚   â”œâ”€â”€ *_genome_view.pdf
â”‚       â”‚   â”œâ”€â”€ *_model_fit.pdf
â”‚       â”‚   â”œâ”€â”€ *_mutations.txt
â”‚       â”‚   â”œâ”€â”€ *_segments.txt
â”‚       â”‚   â”œâ”€â”€ *_sequenza_cp_table.RData
â”‚       â”‚   â”œâ”€â”€ *_sequenza_extract.RData
â”‚       â”‚   â””â”€â”€ *_sequenza_log.txt
â”‚       â””â”€â”€ seqz/                        # ì¤‘ê°„ ìƒì„±ë¬¼
â”‚           â”œâ”€â”€ *.seqz.gz
â”‚           â”œâ”€â”€ *.seqz.gz.tbi
â”‚           â”œâ”€â”€ *.small.seqz.gz
â”‚           â””â”€â”€ *.small.seqz.gz.tbi
â”œâ”€â”€ benchmarks/                           # ì‹¤í–‰ ì„±ëŠ¥ ê¸°ë¡
â”‚   â”œâ”€â”€ sqnz.bam2seqz.<ìƒ˜í”Œ>.txt
â”‚   â”œâ”€â”€ sqnz.binning.<ìƒ˜í”Œ>.txt
â”‚   â””â”€â”€ sqnz.run.<ìƒ˜í”Œ>.txt
â”œâ”€â”€ logs/                                 # ì‹¤í–‰ ë¡œê·¸
â”‚   â”œâ”€â”€ sqnz.bam2seqz.<ìƒ˜í”Œ>.err
â”‚   â”œâ”€â”€ sqnz.bam2seqz.<ìƒ˜í”Œ>.out
â”‚   â”œâ”€â”€ sqnz.binning.<ìƒ˜í”Œ>.err
â”‚   â”œâ”€â”€ sqnz.binning.<ìƒ˜í”Œ>.out
â”‚   â”œâ”€â”€ sqnz.run.<ìƒ˜í”Œ>.err
â”‚   â””â”€â”€ sqnz.run.<ìƒ˜í”Œ>.out
â”œâ”€â”€ pipeline.cmd.sh                       # ì‹¤í–‰ íŠ¸ë¦¬ê±° ìŠ¤í¬ë¦½íŠ¸
â””â”€â”€ samples.config.yaml                   # ì‹¤í–‰ì— í•„ìš”í•œ Config
```

    
# ì‹¤í–‰ ë°©ë²•
    
   1. íŠ¹ì • ìœ„ì¹˜ì— ë¶„ì„ì— í•„ìš”í•œ í´ë”ë¥¼ ìƒì„±í•©ë‹ˆë‹¤. ë¶„ì„ì´ ì‹¤í–‰ë  í´ë”
   2. Sequenza ìˆ˜í–‰ì— í•„ìš”í•œ Normal / Tumor Bam íŒŒì¼ì„ ì¤€ë¹„í•©ë‹ˆë‹¤. ***(Tumor Onlyë¡œëŠ” ë™ì‘í•˜ì§€ ì•ŠìŠµë‹ˆë‹¤.)***
   3. samples.config.yaml íŒŒì¼ì„ ë³µì‚¬í•˜ì—¬, ì•„ë˜ì™€ í˜•ì‹ì— ë§ì¶° ìˆ˜ì •í•©ë‹ˆë‹¤.
        
        ```yaml
        samples:
          {sample_id}: #ìƒ˜í”Œëª…ì´ ë“¤ì–´ê°ˆ ìœ„ì¹˜ ì…ë‹ˆë‹¤.
            normal_bam: {normal_bam_file_path} #normal bam íŒŒì¼ì— ëŒ€í•œ ê²½ë¡œê°€ ì…ë ¥ë  ìœ„ì¹˜ ì…ë‹ˆë‹¤.
            tumor_bam: {tumor_bam_file_path} #tumor bam íŒŒì¼ì— ëŒ€í•œ ê²½ë¡œê°€ ì…ë ¥ë  ìœ„ì¹˜ ì…ë‹ˆë‹¤.
          10_post_rt_leg_fb_3-2: (ì˜ˆì‹œ)
            normal_bam: /data/temp_intern/intern/woo/flagship_figure/bam/DB10/10_PRL3-1_001B8-E10.s.md.ir.bam
            tumor_bam: /data/temp_intern/intern/woo/flagship_figure/bam/DB10/10_post_rt_leg_fb_3-2.s.md.ir.bam
        ```
        
   4. pipline.cmd.sh íŒŒì¼ì„ ë³µì‚¬í•˜ì—¬, ì•„ë˜ì™€ ê°™ì´ ìˆ˜ì •í•©ë‹ˆë‹¤.
        
        ```bash
        wkdir="/data/SJ/projects/flagship_figure/analysis_pipeline" #ë¶„ì„ì´ ì§„í–‰ë  ê²½ë¡œ - ìˆ˜ì • í•„ìš”
        conffn="/data/SJ/projects/flagship_figure/analysis_pipeline/samples.config.yaml" #ì•ì„œ ì‘ì„±í•œ config íŒŒì¼ì´ ìœ„ì¹˜í•œ ê²½ë¡œ - configì™€ ë¶„ì„ ì§„í–‰ ê²½ë¡œê°€ ë‹¬ë¼ë„ ë¬´ë°©í•©ë‹ˆë‹¤. - ìˆ˜ì • í•„ìš”
        
        mkdir -p ${wkdir}/logs
        
        #ì¼ë°˜ì  ê²½ìš° í•˜ê¸° ë¶€ë¶„ ìˆ˜ì •í•  í•„ìš” ì—†ìŒ.
        cmd="snakemake \
            --cores all \
            --jobs 10
            --printshellcmds \
            --snakefile /data/SJ/pipeline/sequenza_pipeline/Snakefile \
            --config 'ref=/data/SJ/pipeline/sequenza_pipeline/refs/ref.yaml' \
            --max-jobs-per-second 5 \
            --configfile ${conffn} \
            --directory ${wkdir} \
            --executor slurm --default-resources slurm_account=pipeline slurm_partition=cpu --slurm-init-seconds-before-status-checks=60 \
            --use-singularity \
            --singularity-args '--bind /data,/tmp'"
        echo $cmd
        ```
        
   5. conda activate snakemake : snakemake ì‹¤í–‰ì— í•„ìš”í•œ conda í™˜ê²½ activate
   6. bash pipline.cmd.sh | /bin/bash : bash shell ì‹¤í–‰ í›„, ì¶œë ¥ë˜ëŠ” ìŠ¤í¬ë¦½íŠ¸ í™•ì¸ í›„ | /bin/bashë¡œ ì‹¤ì œ ì‹¤í–‰í•  ê²ƒ.

# ì‹¤í–‰ í™”ë©´

![image.png](image%201.png)

- ì‹¤í–‰ ì‹œ, ë‹¤ìŒê³¼ ê°™ì´ í˜„ì¬ ì§€ì •ë˜ì–´ ìˆëŠ” config / file_pathì— ëŒ€í•´ì„œ ì¶œë ¥ í›„ ì§„í–‰.
- ìƒ˜í”Œëª… ë“±ì´ ì˜ëª»ë˜ì§€ ì•Šì•˜ëŠ”ì§€ ë§ˆì§€ë§‰ìœ¼ë¡œ í•œ ë²ˆ ë” í™•ì¸í•  ê²ƒ.
- squeueë¥¼ ì‚¬ìš©í•˜ì—¬ ì‘ì—… í™•ì¸ ì‹œ, ìœ„ SLURM run IDë¥¼ ì°¸ì¡°í•  ê²ƒ.
